### DISCUSSION TITLE
extract sub key data without iterating

### DISCUSSION THREAD

**Discussion by JGriff-dev** [nlohmann/json#2474] [Show and tell]
I'm new to nlohmann json and I would first like to congratulate the developer on the best JSON library I've seen, its great!

I've looked for and seen requests for multiple key type functionality, which is something I use a lot. Where a hierarchical object structure is known, accessing it easily & safely can be tedious. I discovered today a very simple way of doing this with this library which I think would merit adding to the documentation. I parse an incoming JSON string containing and object and can extract an array of double values in a sub key using this straight forward method:-

    json jData = json::parse(JSON_String);

    if (jData["DATA"]["Temperature"].is_array())
    {
        std::list<double> dCentigrade =jData["DATA"]["Temperature"];
    }

Very simple & safe without having to iterate though each layer of the JSON. The same "is_xxxxxxx()" method can be used to verify any known sub key.

Thanks
John.

**Comment by nlohmann** [comment]
Glad the implicit conversions made this straightforward - assigning jData["DATA"]["Temperature"] straight into a std::list<double> is exactly the intended usage, and it's covered in the implicit/explicit conversions documentation. Thanks for sharing and for the kind words!

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
